src/api/java/jni/api_solver.h

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Daniel Larraz
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2025 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * The cvc5 Java API.
+ */
+
+#ifndef CVC5__API_SOLVER_H
+#define CVC5__API_SOLVER_H
+#include <cvc5/cvc5.h>
+#include <jni.h>
+
+class ApiSolver : public cvc5::Solver
```

Comment:
Nice, I like this idea of inheriting from public Solver. Perhaps we can apply this to Oracle and Plugin.

Reply:
The application of this idea to `IOracle` and `AbstractPlugin` is not straightforward. `IOracle` is an interface, not a class. `AbstractPlugin`, on the other hand, is a class, so it could store a pointer. However, this would be a breaking change. `AbstractPlugin` is designed to be extended by users, so adding a new field could conflict with user-defined fields. I would postpone this change for now.